### PR TITLE
Remove requirements.txt as a dependency

### PR DESCRIPTION
**Pull request by sverchdotgov**
This converts both docker and tox to use poetry, eliminating usage of requirements.txt in both flows.

- In tox, uses the tox-poetry package which installs dependencies from the lockfile.
- In docker, uses https://stackoverflow.com/questions/53835198/integrating-python-poetry-with-docker as a reference.

Fixes: https://github.com/usds/justice40-tool/issues/651

### CODE REVIEW

**Comment by sverchdotgov** [comment]
Marking this as ready for review. I think the checks are currently failing because of https://github.com/usds/justice40-tool/issues/1108.

**Comment by sverchdotgov** [comment]
Hmm, unfortunately [liccheck](https://pypi.org/project/liccheck/) depends on the existence of requirements.txt. Maybe there's a way to generate it during the tox run.

**Comment by sverchdotgov** [comment]
@esfoobar-usds @saran-ahluwalia I tested this build with the fixes for the "pandas assigning to a copy of a slice issue" and everything passed, so once that's merged in this should work.

**Comment by sverchdotgov** [comment]
@esfoobar-usds @saran-ahluwalia Seems like this is passing now! Does this look good? The most unpleasant thing here is the hack to get liccheck to work since it looks at requirements.txt, but at least that will prevent issues with becoming out of sync because it generates it on demand.

File: data/data-pipeline/tox.ini [hunk 1]
```diff
@@ -7,32 +7,17 @@ skip_missing_interpreters = true
...
+allowlist_externals = bash
 commands = safety check
-           liccheck
+           bash scripts/run-liccheck.sh
 
 [testenv:pytest]
 # Run tests
 deps = pytest
-commands = pytest
+commands = pytest --full-trace
```
Comment (esfoobar-usds) [inline data/data-pipeline/tox.ini:23]: What is this flag?

**Review by esfoobar-usds — APPROVED** [review]
🍏  -- just one small question

File: data/data-pipeline/tox.ini (hunk 1 above)
Comment (sverchdotgov) [reply data/data-pipeline/tox.ini:23]: Making the output more verbose on failures: https://docs.pytest.org/en/latest/how-to/output.html#modifying-python-traceback-printing

I ran into https://github.com/usds/justice40-tool/issues/1108 while working on this, and the traceback helped.

**Pull request merged**